scylla-cdc-base/src/test/java/com/scylladb/cdc/model/worker/WorkerTest.java

```diff
@@ -0,0 +1,662 @@
+    ));
+
+    @Test
+    public void testWorkerReadsAnyWindows() {
+        // Worker should start reading windows
+        // from the beginning of the generation.
+
+        MockWorkerTransport workerTransport = new MockWorkerTransport();
+        MockWorkerCQL mockWorkerCQL = new MockWorkerCQL();
+        Consumer noOpConsumer = Consumer.syncRawChangeConsumer(c -> {});
+
+        try (WorkerThread workerThread = new WorkerThread(
+                mockWorkerCQL, workerTransport, noOpConsumer, TEST_GENERATION, TEST_TABLE_NAME)) {
+            DEFAULT_AWAIT.until(() -> mockWorkerCQL.isReaderFinished(generateTask(TEST_GENERATION, 0, TEST_TABLE_NAME,
+                    TEST_GENERATION_START_MS, TEST_GENERATION_START_MS + DEFAULT_QUERY_WINDOW_SIZE_MS)));
+
+            DEFAULT_AWAIT.until(() -> mockWorkerCQL.isReaderFinished(generateTask(TEST_GENERATION, 0, TEST_TABLE_NAME,
+                    TEST_GENERATION_START_MS + DEFAULT_QUERY_WINDOW_SIZE_MS,
+                    TEST_GENERATION_START_MS + 2 * DEFAULT_QUERY_WINDOW_SIZE_MS)));
+
+            DEFAULT_AWAIT.until(() -> mockWorkerCQL.isReaderFinished(generateTask(TEST_GENERATION, 0, TEST_TABLE_NAME,
+                    TEST_GENERATION_START_MS + 2 * DEFAULT_QUERY_WINDOW_SIZE_MS,
+                    TEST_GENERATION_START_MS + 3 * DEFAULT_QUERY_WINDOW_SIZE_MS)));
+
+            DEFAULT_AWAIT.until(() -> mockWorkerCQL.isReaderFinished(generateTask(TEST_GENERATION, 3, TEST_TABLE_NAME,
+                    TEST_GENERATION_START_MS, TEST_GENERATION_START_MS + DEFAULT_QUERY_WINDOW_SIZE_MS)));
+        }
+    }
+
+    @Test
+    public void testWorkerTrimsWindowWithTTL() {
+        // If a TTL was set on a table (as is the case
+        // in Scylla by default on the CDC log table), Worker
+        // should not start reading from the beginning of
+        // generation, but from |now - ttl|.
+
+        MockWorkerTransport workerTransport = new MockWorkerTransport();
+        MockWorkerCQL mockWorkerCQL = new MockWorkerCQL();
+
+        // Setting a 146 second TTL:
+        mockWorkerCQL.setTablesTTL(Collections.singletonMap(TEST_TABLE_NAME, Optional.of(146L)));
+
+        // "Now" will be at second 935:
+        Clock clock = Clock.fixed(Instant.ofEpochSecond(935), ZoneOffset.systemDefault());
+
+        Consumer noOpConsumer = Consumer.syncRawChangeConsumer(c -> {});
+
+        try (WorkerThread workerThread = new WorkerThread(
+                mockWorkerCQL, workerTransport, noOpConsumer, TEST_GENERATION, clock, TEST_TABLE_NAME)) {
+
+            // Expecting to see a window queried from time point (now - ttl) = (935 - 146).
+            // Multiplying by 1000 to convert from seconds to milliseconds.
+            DEFAULT_AWAIT.until(() -> mockWorkerCQL.isReaderFinished(generateTask(TEST_GENERATION, 0, TEST_TABLE_NAME,
+                    (935 - 146) * 1000, (935 - 146) * 1000 + DEFAULT_QUERY_WINDOW_SIZE_MS)));
```

Comment:
I think `935` and `146` should be constants (just as you did in `testWorkerTrimsSavedTaskStatesWithTTL`)

Reply:
Fixed